src/Datasource/EntityTrait.php

```diff
@@ -482,7 +482,7 @@ public function setVirtual(array $properties, $merge = false)
             return $this;
         }
 
-        $this->_virtual += $properties;
+        $this->_virtual = array_merge($this->_virtual, $properties);
```

Comment:
I no saying about duplicate key/value :)
I just sayed there are duplicated a unit test that @lorenzo was fix it in https://github.com/cakephp/cakephp/pull/10364/commits/bb9eb987b519eef1e4809e970093ab2f00782cdd
Thanks @lorenzo 
Why we need array_unique()?